### ISSUE TITLE
[FEATURE REQUEST] automatically format svg files

### ISSUE THREAD

**Issue by Snailedlt** [devicons/devicon#1562]
### I have searched through the issues and didn't find my problem.

- [X] Confirm

### Problem

It's pretty hard to read the code when it's all in a single line. Also there's often a missing newline at the end of the files, which [can cause issues with old code editors.](https://stackoverflow.com/a/2287990/12206312)

### Possible Solution

Add two new github workflows:

1. Automatically format the svg icons to a readable format with on every new commit.
2. When the PR is merged to develop, re-format the icon so it's all in a single line.

Both of these can be done manually in vscode, which means there's probably open-source tools out there that we can use to automate the process :)

For both steps the formatting should include a newline at the end of the file to support old editors

**Comment by Snailedlt** [comment]
@lunatic-fox agreed? :)

**Comment by lunatic-fox** [comment]
Loved it! 🤩

Perhaps we can use SVGO with `pretty: true` and deactivating optimizations for the first workflow and SVGO with default `pretty: false` in the second workflow with all optimizations on.

We still have to take a look about the new line at the end of the file, pretty easy to do with Node, but I want to use as little as possible of the workflow time, and I think SVGO also allows to insert custom code.

**Comment by Snailedlt** [comment]
Hmm, maybe. Haven't looked into it. 

But we need to have optimization done before it's merged to develop, so we can check that it didn't break anything with the icons. That means it needs to be done before step 2 hwre, since step 2 is done when the PR is merged :)

**Comment by Snailedlt** [comment]
Hmm, maybe. Haven't looked into it. 

But we need to have optimization done before it's merged to develop, so we can check that it didn't break anything with the icons. That means it needs to be done before step 2 hwre, since step 2 is done when the PR is merged :)

**Comment by Snailedlt** [comment]
On second thought, minifying the icons into a single file makes them hard to review. So updating icons would still be an issue. Maybe we should just automatically add minified versions of the icons as a part of the  [release strategy](https://github.com/devicons/devicon/wiki/Release-strategy,-conventions,-preparation-and-execution)? If we do that we can edit the original icons more easily, and still have minimized icons for public use

**Comment by Snailedlt** [comment]
@lunatic-fox Excellent idea! Ship it, ship it! 🚢

**Comment by Snailedlt** [comment]
Btw, there's already a github action to add a newline character to all files that are missing it in a PR -> https://github.com/marketplace/actions/newline-action

I made a new issue to add it (#1624), and removed it from this issue.

**Comment by Snailedlt** [comment]
We can definitely use SVGO for formatting btw.
[SVGOMG](https://jakearchibald.github.io/svgomg/) can prettify/format the SVG's:
![image](https://user-images.githubusercontent.com/43886029/211399131-8f6dc2c4-3bd7-4f0d-affb-0294cbb3ff73.png)

I tested it and it works great.

Today I got another idea for how we could go about doing this though!
What if we just make a `.minified` version of all icons, and keep the prettified ones? We would then have for example: `devicon-original.svg` and `devicon-original.minified.svg`. Then we can use the minified version in our font and npm, and still have the pretty one for when the icon needs to be updated. I know a lot of libraries use this strategy for .js and .css files, so it's tried and tested
